packages/cli/src/hooks/init/termination-signals.ts

```diff
@@ -0,0 +1,16 @@
+import {buildEvent} from '../analytics/eventUtils';
+
+export const signals = ['SIGINT', 'SIGTERM', 'SIGQUIT', 'SIGHUP'] as const;
+
+export const handleTerminationSignals = () => {
+  signals.forEach((signal) => process.on(signal, () => trackEvent(signal)));
+};
+
+function trackEvent(termination_signal: string) {
+  config.runHook('analytics', {
+    event: buildEvent('interrupted operation', {termination_signal}),
+  });
+
+  // eslint-disable-next-line no-process-exit
+  process.exit(0);
```

Comment:
In case of termination, the exitCode should be `128+n` with n being the ' Signal numbering for standard signals'
Luckily, the second args of the signal event is `n`.

See:
 - https://nodejs.org/api/process.html#exit-codes
 - https://man7.org/linux/man-pages/man7/signal.7.html